Sources/Sugar/Authentication/JWTAuthenticationMiddleware.swift

```diff
@@ -34,12 +33,13 @@ public final class JWTAuthenticationMiddleware<A: JWTAuthenticatable>: Middlewar
         }
 
         guard let bearer = req.http.headers.bearerAuthorization else {
-            return try next.respond(to: req)
+            throw AuthenticationError.missingToken
```

Comment:
@steffendsommer can you think of any reason not to throw here?

Reply:
I made the change because this is the only middleware that i'm using and without that, the middleware is useless at least for me 😅

Reply:
@siemensikkema @gperdomor I think *not* throwing is more in line with the other auth middlewares coming from [Auth](https://github.com/vapor/auth). I think this strategy makes sense, since you might have endpoints that has optional auth. This basically leaves it up to the callback of the route to determine if auth is required (by using `requireAuthenticated`).

Reply:
@gperdomor Can you elaborate a bit on why it's useless without the `throw` here?

Reply:
@steffendsommer In my case i only want to validate the token without authenticate with the database, i always assume the token info as correct... But also understand the case you mention or even having a chain of middlewares... Said that, how can i handle my case without throwing this error?

Reply:
@gperdomor I might be misunderstanding your use case, but could this be handled by setting `shouldAuthenticate` to false when setting up the middleware? https://github.com/nodes-vapor/sugar/blob/master/Sources/Sugar/Authentication/JWTAuthenticationMiddleware.swift#L49

Reply:
Yes, `shouldAuthenticate = false` works for me, but also need the precense of the bearer token to avoid the execution of the controller.

In conclusion:
- I need force the token presence, the token should be valid and `shouldAuthenticate = false`, run controller function
- No token presence or invalid token, prevent the controller function execution.

Maybe we can add a conditional to throw the error when no token is present... Something like `forceToken = true`?

Reply:
@gperdomor I apologise for the delayed reply. I understand your case and I do think it's common. Although I do see the convenience in having this flag on the middleware, I don't think it's in line with how the other auth middlewares are made. Please see this PR for how we handle this in JWT Keychain: https://github.com/nodes-vapor/jwt-keychain/pull/121